File: PocketBeagle2/light_sensor/rust/src/main.rs
use std::thread::sleep;
use std::time::Duration;

use beagle_linux_sdk::boards::pocketbeagle2::P1_19;
use beagle_linux_sdk::LightSensor;

fn main() {
    let mut ldr = LightSensor::new(P1_19, None).unwrap();

    loop {
        if ldr.value().unwrap() {
            println!("Light");
        } else {
            println!("Dark");
        }

        sleep(Duration::from_millis(500));
    }
}

File: PocketBeagle2/light_sensor/python/main.py
from time import sleep
from beagle_linux_sdk import LightSensor
from beagle_linux_sdk.boards.pocketbeagle2 import P1_19

ldr = LightSensor(P1_19)

while True:
    if ldr.value():
        print("Light")
    else:
        print("Dark")
    sleep(0.5)
